### PR TITLE
route call optimize

### PR DESCRIPTION
**Pull request by wangxw666**
in `RegistryDirectory.List`
```
routerChain := dir.RouterChain()

	if routerChain == nil {
		dir.invokersLock.RLock()
		defer dir.invokersLock.RUnlock()
		return dir.cacheInvokers
	}
```
 
If I didn't create the route it should just go back `cacheInvokers`，but  `NewRegistryDirectory` created a default routerChain struct, so the judgment here will never be valid

### CODE REVIEW

**Review by AlexStocks — APPROVED** [review]

**Comment by wangxw666** [comment]
image

image

image

**Comment by AlexStocks** [comment]
@LaurenceLiZhixin @zhaoyunxing92

**Review by LaurenceLiZhixin — APPROVED** [review]

File: cluster/directory/base_directory.go [hunk 1]
```diff
@@ -44,15 +43,11 @@ type BaseDirectory struct {
...
-		url:         url,
-		destroyed:   atomic.NewBool(false),
-		routerChain: &chain.RouterChain{},
+func NewBaseDirectory(url *common.URL) BaseDirectory {
+	return BaseDirectory{
+		url:       url,
+		destroyed: atomic.NewBool(false),
 	}
-	// start to listen notify
-	go dir.routerChain.Loop()
```
Comment (zouyx) [inline cluster/directory/base_directory.go:54]: Why will you delete this code? if you delete this code , router will not effect no longer

Comment (wangxw666) [reply cluster/directory/base_directory.go:54]: 因为baseDirectory在new的时候不需要创建routerChain，因为我没有创建任何路由策略，所不需要loop

**Review by zhaoyunxing92 — APPROVED** [review]
approved these changes

**Pull request merged**